src/main/java/com/uber/cadence/internal/replay/ReplayDecisionTaskHandler.java

```diff
@@ -98,63 +109,110 @@ private Result handleDecisionTaskImpl(DecisionTaskWithHistoryIterator decisionTa
       throws Throwable {
     HistoryHelper historyHelper = new HistoryHelper(decisionTaskIterator);
     PollForDecisionTaskResponse decisionTask = historyHelper.getDecisionTask();
-    ReplayDecider decider = createDecider(decisionTask);
-
-    if (decisionTask.isSetQuery()) {
-      RespondQueryTaskCompletedRequest queryCompletedRequest =
-          new RespondQueryTaskCompletedRequest();
-      queryCompletedRequest.setTaskToken(decisionTask.getTaskToken());
-      try {
-        byte[] queryResult = decider.query(historyHelper, decisionTask.getQuery());
-        queryCompletedRequest.setQueryResult(queryResult);
-        queryCompletedRequest.setCompletedType(QueryTaskCompletedType.COMPLETED);
-      } catch (Throwable e) {
-        // TODO: Appropriate exception serialization.
-        StringWriter sw = new StringWriter();
-        PrintWriter pw = new PrintWriter(sw);
-        e.printStackTrace(pw);
-        queryCompletedRequest.setErrorMessage(sw.toString());
-        queryCompletedRequest.setCompletedType(QueryTaskCompletedType.FAILED);
+
+    ReplayDecider decider =
+        stickyTaskListName == null
+            ? createDecider(decisionTask)
+            : cache.getOrCreate(decisionTask, this::createDecider);
```

Comment:
I'm not sure I like the notion of a disabled cache. The ReplayDecisionTaskHandler currently is responsible for responding back to the server with StickyAttributes. So it makes sense that it would also be responsible for choosing whether to cache workflows or not (and not push that decision on the cache).